robotoff/app/core.py

```diff
@@ -165,6 +173,38 @@ def get_predictions(
         return query.iterator()
 
 
+def get_image_predictions(
+    with_logo: Optional[bool] = False,
+    barcode: Optional[str] = None,
+    type: Optional[str] = None,
+    offset: Optional[int] = None,
+    count: bool = False,
+    limit: Optional[int] = 25,
+) -> Iterable[LogoAnnotation]:
+
+    query = LogoAnnotation.select().join(ImagePrediction).join(ImageModel)
+
+    where_clauses = []
+
+    if barcode:
+        where_clauses.append(ImageModel.barcode == barcode)
+
+    if type:
+        where_clauses.append(LogoAnnotation.annotation_type == type)
+
+    if not with_logo:
+        # return only images without logo
+        query = query.where(ImagePrediction.image.is_null())
```

Comment:
you have to use LefOuter join as for images but joining on LogoAnnotation this time.